Sources/HAP/Endpoints/pairVerify().swift

```diff
@@ -63,7 +63,45 @@ func pairVerify(device: Device) -> Responder {
         } catch {
             logger.warning("Could not verify pairing: \(error)")
             setSession(for: context, to: nil)
-            return .badRequest
+
+            // HAP Specification 4.8.4:
+            // Must return authentication error to Verify Finish Response (M4) when
```

Comment:
The documentation you've quoted says that the authentication error should be returned for "Verify Finish Response (M4)", but the implementation will now also do that for the start request. Which is broader than the M4 step. I suppose this is not a problem?

Reply:
This is not an issue, none of the errors caught are thrown in startRequest, so there is no change to behavior during earlier steps. Only the following errors are caught and result in the authentication error response.
                PairVerifyController.Error.couldNotDecrypt,
                PairVerifyController.Error.couldNotDecode,
                PairVerifyController.Error.noPublicKeyForUser,
                PairVerifyController.Error.invalidSignature: